subt/main.py

```diff
@@ -141,10 +142,43 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self.robot.unregister(self.callback)
 
 
+class TimeoutEx(Exception):
+    pass
+
+
+class TimeoutMonitor:
```

Comment:
I would vote for separate file (`main.py` is too big already) ... and maybe it is generic enough that we want it in `osgar.lib`? Or monitors??